### PR TITLE
Fix bug where algorithm names weren't being logged properly in traces

### PR DESCRIPTION
**Pull request by coryMosaicML**

### CODE REVIEW

**Comment by coryMosaicML** [comment]
Turns out some of our tests rely on things being done the previous way, hence engine tests now fail, and this could cause some strange behaviors if the same algorithm is used more than once as in the tests. So, this will take some more work.

**Comment by hanlint** [comment]
Nice -- has this now exposed a passing test is actually failing?

**Comment by coryMosaicML** [comment]
No, things work correctly on current dev, only issue is some of our docstrings are misleading, and changing code to be in line with the docs causes things to break.

**Comment by coryMosaicML** [comment]
Closing this for now until I fix it properly.

File: composer/core/engine.py [hunk 1]
```diff
@@ -144,7 +146,7 @@ def _setup_trace(algorithms: Sequence[Algorithm], event: Event) -> Traces:
     The keys are of format ``<algorithm_name>/<event>`` (e.g.,  ``Blurpool/INIT``) and values are an instance of
     :class:`Trace`.
     """
-    return OrderedDict([(f'{algo}/{event}', Trace()) for algo in algorithms])
+    return OrderedDict([(f'{algo}/{event}', Trace(name=algo.__class__.__name__)) for algo in algorithms])
```
Comment (ravi-mosaicml) [inline composer/core/engine.py:149]: I don't think the `name` needs to be added to the trace, but rather, `{algo}` should be `{type(alg).__name__}`

**Pull request closed without merging**